src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -81,7 +81,7 @@ struct ExpressionAnalyzerData
   *
   * NOTE: if `ast` is a SELECT query from a table, the structure of this table should not change during the lifetime of ExpressionAnalyzer.
   */
-class ExpressionAnalyzer : protected ExpressionAnalyzerData, private boost::noncopyable, protected WithContext
+class ExpressionAnalyzer : public ExpressionAnalyzerData, private boost::noncopyable, protected WithContext
```

Comment:
I don't think you need to change public interface of `ExpressionAnalyzer` for this optimization.

Reply:
Done.